gradle-jdks-certs/src/main/resources/gradle-jdk-resolver.sh

```diff
@@ -0,0 +1,128 @@
+#     and the distribution_url=`gradle/jdks/${majorVersion}/${os}/${arch}/download_url.txt`
+#   (2) Installs the distribution in `$HOME/.gradle/gradle-jdks/${local_path}` based on the
+#     local_path=`gradle/jdks/${majorVersion}/${os}/${arch}/local_path.txt`
+#   (3) Calls the java class `JdkSpecCertSetup` that will set up the certificates for the locally installed distribution
+#   (4) Sets up the JAVA_HOME env variable to the currently installed JDK
+#
+#
+#   Important for running:
+#   This script requires all of these POSIX shell features:
+#         * functions;
+#         * expansions «$var», «${var}», «${var%suffix}», and «$( cmd )»;
+#         * various built-in commands including «command» and «set».
+#
+##############################################################################
+
+tmp_work_dir=$(mktemp -d)
+
+die () {
+    echo
+    echo "$*"
+    echo
+    # TODO(crogoz): Cleaning up
+    echo rm -rf "$tmp_work_dir"
+    exit 1
+} >&2
+
+# OS specific support
+case "$( uname )" in                          #(
+  Linux* )          os_name="linux"  ;;       #(
+  Darwin* )         os_name="macos"  ;;       #(
+  * )               die "ERROR Unsupported OS: $( uname )" ;;
+esac
+
+if [ "$os_name" = "linux" ]; then
+    if [ -f /etc/alpine-release ]; then
+      os_name="alpine-linux"
+    fi
+fi
+
+# Arch specific support, see: gradle-jdks:com.palantir.gradle.jdks.CurrentArch.java
+case "$(uname -m)" in                         #(
+  x86-64* )       arch_name="x86-64"  ;;      #(
+  x64* )          arch_name="x86-64"  ;;      #(
+  amd64* )        arch_name="x86-64"  ;;      #(
+  arm64* )        arch_name="aarch64"  ;;     #(
+  arm* )          arch_name="aarch64"  ;;     #(
+  aarch64* )      arch_name="aarch64"  ;;     #(
+  x86* )          arch_name="x86"  ;;         #(
+  i686* )         arch_name="x86"  ;;         #(
+  * )             die "ERROR Unsupported architecture: $( uname -m )" ;;
+esac
+
+read -r major_version < gradle/gradle-jdk-major-version.txt
+read -r distribution_url < gradle/jdks/"$major_version"/$os_name/$arch_name/download-url.txt
+read -r distribution_local_path < gradle/jdks/"$major_version"/$os_name/$arch_name/local-path.txt
+
+regex_pattern="^([a-zA-Z0-9_-]+)-([0-9.]+)-([a-z0-9]+)$"
+
+# Check if the input string matches the regex pattern
+if echo "$distribution_local_path" | grep -q -E "$regex_pattern"; then
+  name=$(echo "$distribution_local_path" | sed -n -E "s/$regex_pattern/\1/p")
+  version=$(echo "$distribution_local_path" | sed -n -E "s/$regex_pattern/\2/p")
+  hash=$(echo "$distribution_local_path" | sed -n -E "s/$regex_pattern/\3/p")
+else
+  die "ERROR: $distribution_local_path for gradle-jdk-major-version=$major_version and $(uname -m) is invalid, expected to match pattern: $regex_pattern"
+fi
+
+# Check if distribution exists in $HOME/.gradle/gradle-jdks
+jdk_installation_directory=$HOME/.gradle/gradle-jdks/$name-$version-$hash
```

Comment:
One unfortunate thing we're going to have to deal with is concurrency. Someone could run different two `./gradlew` invocations at the same time and they both try to write to the directory, possibly causing badness. 

We suffer this problem in the current java implementation:
* https://github.com/palantir/gradle-jdks/pull/71
* https://github.com/palantir/gradle-jdks/pull/73
* https://github.com/palantir/gradle-jdks/pull/77

I think we're going to need to do something similar in shell script here. I did some basic research of options for file based looking and it looks like: 
* [`shlock`](https://www.unix.com/man-page/osx/1/shlock/) is installed on macos
* [`flock`](https://manpages.ubuntu.com/manpages/xenial/man1/flock.1.html) is installed most of the time on linux (as per usual alpine has the [busybox version of flock](https://man.archlinux.org/man/busybox.1.en#flock) which has fewer features)
* otherwise we can use `mv` and hope it is atomic (no really much else we can do).

`flock` > `shlock` > `mv`, so maybe we just see what's installed 

We also need to do [double checked locking](https://en.wikipedia.org/wiki/Double-checked_locking) so we don't take out the lock unnecessarily / another process hasn't done everything before we actually get the lock.

Tbh, this is one case where I'm not sure _how much_ lock safety we need - unlike the java version we don't have multiple threads in the same process all possibly trying to install the same JDK, but we still have the races between `./gradlew` invocations.